### ISSUE TITLE
Evaluate how images are resized in the repo

### ISSUE THREAD

**Issue by tlpss** [airo-ugent/airo-mono#89]
Recently I've discovered that `cv2.resize` and `PIL.Image.resize` have quite different implementations. 

In particular, opencv does not scale the size of the filter used in downscaling (i.e. even if you downscale 100 times, only the neighbouring pixels are used to calculate the interpolation), which results in aliasing artifacts. Most notably, sharp edges will have 'staircase' artifacts, which increase with the size of the downscaling. (funny sidenote downscaling twice 2x will give better results than downscaling once 4x). Pillow uses adaptive filter size, which avoids aliasing by 'smoothing' the image. The downscaled image will contain no artifacts but will be a bit 'blurred'.

More information: https://arxiv.org/pdf/2104.11222.pdf, https://zuru.tech/blog/the-dangers-behind-image-resizing
illustration:
![image](https://github.com/airo-ugent/airo-mono/assets/37955681/e153c633-72f7-420b-beff-5dc47508425b)

This also impacts Deep learning training. I've found that mAP can _differ up to 5%_  when resizing with PIL on the train set and with cv2 on the test set versus using the same resize method for both (downscaling with factor 4).

Long story short, we should probably make sure that we consistently use the same downsampling method as much as possible in the repo to make sure that the artifacts do not influence inference/test performance.

I've now used **PILLOW's bicubic interpolation** in the coco tools and suggest to set that as default throughout the repo.

@Victorlouisdg, the image_transforms in the camera toolkit is the first place in the repo that comes to mind but there might be other places where we downscale images.

**Comment by Victorlouisdg** [comment]
I ran some quick performance tests. OpenCV is at least **10x faster** than PIL at downscaling from (1920, 1080) to (320, 240).  Because the `ImageTransform`s are also intended for real-time visualization, I'll likely leave OpenCV as the default. However, I have to investigate the quality concerns further. Possible actions:

- Use `cv2.INTER_BICUBIC` instead of the default `cv2.INTER_LINEAR` because speed is similar
- Add constructor argument to `Resize` class to select between OpenCV en PIL resize implementation.

![image](https://github.com/airo-ugent/airo-mono/assets/23340946/32b1daa3-a7bb-496c-8ba3-02df83d3c29b)

**Comment by Victorlouisdg** [comment]
Notebook to run the comparison: https://gist.github.com/Victorlouisdg/27c22053e9c9be2a71f2912aaecb3431

**Comment by m-decoster** [comment]
As far as I can tell, there's no code path in airo-mono itself that uses `ImageTransform`, so in my opinion we could replace it with Pillow's resize to be consistent with the airo-dataset-tools implementation.

We could add a `FastResize` as an alternative with clear documentation for users to use that one for real-time visualisations?
